lib/spack/spack/test/conftest.py

```diff
@@ -259,6 +259,17 @@ def _verify_executables_noop(*args):
     return None
 
 
+def _host():
+    """Mock archspec host so there is no inconsistency on the Windows platform
+    This function cannot be local as it needs to be pickleable"""
+    return archspec.cpu.Microarchitecture("x86_64", [], "generic", [], {}, 0)
```

Comment:
This line causes test failures on M1

@scheibelp @johnwparent you may be interested in this.

Reply:
@scheibelp I think this means it's probably time to revisit the whole "Spack has multiple entry points for platform/arch detection and they conflict at test time" issue that was raised [here](https://github.com/spack/spack/pull/33385#discussion_r1040036318)

Reply:
Thanks for the ping @becker33, feel free to tag my/reach out to iterate on a solution for your issue. Do you have a link to a CI run with this failure?